src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -31,6 +34,8 @@ public User register(String email, String password) {
         User user = new User();
         user.setEmail(email);
         user.setPassword(password);
-        return userService.add(user);
+        User userFromDb = userService.add(user);
+        shoppingCartService.registerNewShoppingCart(userFromDb);
+        return userFromDb;
```

Comment:
```suggestion
         userService.add(user);
        shoppingCartService.registerNewShoppingCart(user);
        return user;
```
Should work this way too